xlators/cluster/afr/src/afr-common.c

```diff
@@ -7876,3 +7876,62 @@ afr_ta_dict_contains_pending_xattr(dict_t *dict, afr_private_t *priv, int child)
 
     return _gf_false;
 }
+
+gf_boolean_t
+afr_is_outcast_set(dict_t *xdata, int type)
+{
+    int idx = -1;
+    void *pending_raw = NULL;
+    int *pending_int = NULL;
+
+    idx = afr_index_for_transaction_type(type);
+
+    if (dict_get_ptr(xdata, AFR_OUTCAST_DEFAULT, &pending_raw) == 0) {
+        if (pending_raw) {
+            pending_int = pending_raw;
+
+            if (pending_int[idx]) {
+                /* We are not using ntoh32 here since it is just a non-zero
+                 * check
+                 */
+                return _gf_true;
+            }
+        }
+    }
+    return _gf_false;
+}
+
+gf_boolean_t
+afr_is_any_outcast_set(dict_t *xdata)
+{
+    int idx = -1;
+    void *pending_raw = NULL;
+    int *pending_int = NULL;
+    afr_transaction_type type;
+
+    type = AFR_METADATA_TRANSACTION;
+    idx = afr_index_for_transaction_type(type);
+
+    /* TODO
+     * Currently outcast logic is only implemeted for meta-data transactions,
+     * When we do the outcase for all the fop this has to be changed to a loop
+     * to check for all types of fops.
+     */
+    if (dict_get_ptr(xdata, AFR_OUTCAST_DEFAULT, &pending_raw) == 0) {
+        if (pending_raw) {
+            pending_int = pending_raw;
+
+            if (pending_int[idx]) {
+                /* We are not using ntoh32 here since it is just a non-zero
+                 * check
+                 */
+                return _gf_true;
+            }
+        }
+    }
+    /*
+     * At the moment, outcast for AFR_METADATA_TRANSACTION is
+     * implemented
+     */
```

Comment:
Both afr_is_outcast_set() and afr_is_any_outcast_set() are doing the same thing. Can you refactor so that the latter calls the former?